### PR TITLE
Reorganization code

### PR DESCRIPTION
**Pull request by jaanonim**

### CODE REVIEW

File: app/cruds/posts.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+import hashlib
```
Comment (n2oneProgrammer) [inline app/cruds/posts.py]: useless

File: app/cruds/posts.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+from fastapi import UploadFile
```
Comment (n2oneProgrammer) [inline app/cruds/posts.py]: useless

File: app/cruds/posts.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+from sqlalchemy import and_, or_
+from sqlalchemy.orm import Session
+
+from .. import models
+from ..file_service import reupload_image, upload_image
```
Comment (n2oneProgrammer) [inline app/cruds/posts.py]: useless

File: app/cruds/posts.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+from ..schemats import posts, users
```
Comment (n2oneProgrammer) [inline app/cruds/posts.py]: users is useless

File: app/cruds/users.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import hashlib
+
+from fastapi import UploadFile
+from sqlalchemy import and_, or_
```
Comment (n2oneProgrammer) [inline app/cruds/users.py]: or_ is useless

File: app/dependencies.py [hunk 6]
```diff
@@ -45,7 +44,7 @@ async def get_current_user(
         username: str = payload.get("sub")
         if username is None:
             raise credentials_exception
-        token_data = schema.TokenData(username=username)
+        token_data = TokenData(username=username)
     except JWTError:
         raise credentials_exception
     user = crud.get_user_by_email(db, token_data.username)
```
Comment (n2oneProgrammer) [inline app/dependencies.py]: this crud is not implemented

File: app/routers/posts.py [hunk 7]
```diff
@@ -1,14 +1,12 @@
...
+from fastapi import APIRouter, Depends, HTTPException
 from sqlalchemy.orm import Session
-from typing import Optional
 
-from .. import crud
-from ..dependencies import get_db
-from ..models import AccountType, User, PostType
-from ..schema import PostResponse, PostCreate, PostEdit
-from ..dependencies import get_current_user
+from ..cruds import posts
```
Comment (n2oneProgrammer) [inline app/routers/posts.py]: you dont change name of crud in line 29 and 45,75,

File: app/routers/posts.py [hunk 8]
```diff
@@ -1,14 +1,12 @@
...
+from ..dependencies import get_current_user, get_db
+from ..models import AccountType, PostType, User
+from ..schemats.posts import PostCreate, PostEdit, PostResponse
```
Comment (n2oneProgrammer) [inline app/routers/posts.py:10]: idk why, but have_user_permission is not imported

File: app/routers/users.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+from sqlalchemy.orm import Session
+
+router = APIRouter(tags=["user"])
+
+
+@router.get("/users", response_model=List[users.UserResponse])
+async def get_organizations(
+    skip: int = 0, limit: int = 100, db: Session = Depends(get_db)
+):
+    users = crud.get_users_by_type(db, AccountType.ORGANIZATION, skip, limit)
```
Comment (n2oneProgrammer) [inline app/routers/users.py]: crud not implemented

File: app/routers/users.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    },
+)
+async def get_all_users(
+    skip: int = 0,
+    limit: int = 100,
+    db: Session = Depends(get_db),
+    current_user: users.User = Depends(get_current_user),
+):
+    have_user_permission(current_user, [AccountType.ADMIN])
+    users = crud.get_users(db, skip, limit)
```
Comment (n2oneProgrammer) [inline app/routers/users.py]: crud not implemented

File: app/routers/users.py [hunk 11]
```diff
@@ -0,0 +1,144 @@
...
+    return current_user
+
+
+@router.get(
+    "/users/{user_id}",
+    response_model=users.UserResponse,
+    responses={404: {"description": "User not found"}},
+)
+async def get_user_by_user_id(user_id: int, db: Session = Depends(get_db)):
+    user = crud.get_user_by_userid(db, user_id)
```
Comment (n2oneProgrammer) [inline app/routers/users.py]: crud not implemented

File: app/routers/users.py [hunk 12]
```diff
@@ -0,0 +1,144 @@
...
+@router.patch("/users/me", response_model=users.UserResponse)
+async def edit_user_me(
+    userData: users.UserEditMe = Depends(users.UserEditMe.as_form),
+    new_avatar: UploadFile = File(default=None),
+    db: Session = Depends(get_db),
+    current_user: users.User = Depends(get_current_user),
+):
+    data = UserEdit(**userData.dict())
+    try:
+        user = await crud.update_user(db, current_user, data, new_avatar)
```
Comment (n2oneProgrammer) [inline app/routers/users.py:95]: crud not implemented

File: app/routers/users.py [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+
+@router.patch("/users/me", response_model=users.UserResponse)
+async def edit_user_me(
+    userData: users.UserEditMe = Depends(users.UserEditMe.as_form),
+    new_avatar: UploadFile = File(default=None),
+    db: Session = Depends(get_db),
+    current_user: users.User = Depends(get_current_user),
+):
+    data = UserEdit(**userData.dict())
```
Comment (n2oneProgrammer) [inline app/routers/users.py:93]: USerEdit not implemented

File: app/schemats/posts.py [hunk 14]
```diff
@@ -0,0 +1,66 @@
+from datetime import datetime
+from decimal import Decimal
+from typing import List, Optional
```
Comment (n2oneProgrammer) [inline app/schemats/posts.py]: LIst is useless

File: app/schemats/posts.py [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+
+from app.models.account_type import AccountType
```
Comment (n2oneProgrammer) [inline app/schemats/posts.py]: useless

File: app/schemats/posts.py [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+from app.models.post import PostType
+from fastapi import Form, Query
```
Comment (n2oneProgrammer) [inline app/schemats/posts.py]: useless

File: app/schemats/posts.py [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+from pydantic import BaseModel
+
+# from .users import User
```
Comment (n2oneProgrammer) [inline app/schemats/posts.py]: ok funny comment

File: app/schemats/users.py [hunk 18]
```diff
@@ -2,64 +2,17 @@
 from decimal import Decimal
```
Comment (n2oneProgrammer) [inline app/schemats/users.py]: Useless

**Review by n2oneProgrammer — CHANGES_REQUESTED** [review]

File: app/schemats/posts.py (hunk 17 above)
Comment (jaanonim) [reply app/schemats/posts.py]: This is a problem because we have there recursive importing. Idk how to fix it. :-(

**Review by n2oneProgrammer — APPROVED** [review]
LGTM

**Pull request merged**